src/tools/ScenarioMeasurement/Startup/TimeToMainParser.cs

```diff
@@ -0,0 +1,90 @@
+﻿using Microsoft.Diagnostics.Tracing;
+using Microsoft.Diagnostics.Tracing.Parsers;
+using Microsoft.Diagnostics.Tracing.Session;
+using Reporting;
+using System;
+using System.Collections.Generic;
+
+namespace ScenarioMeasurement
+{
+    internal class TimeToMainParser : IParser
+    {
+        public void EnableKernelProvider(TraceEventSession kernel)
+        {
+            kernel.EnableKernelProvider((KernelTraceEventParser.Keywords)(KernelTraceEventParser.Keywords.Process | KernelTraceEventParser.Keywords.Thread | KernelTraceEventParser.Keywords.ContextSwitch));
+        }
+
+        public void EnableUserProviders(TraceEventSession user)
+        {
+            user.EnableProvider(ClrPrivateTraceEventParser.ProviderGuid, TraceEventLevel.Verbose, (ulong)ClrPrivateTraceEventParser.Keywords.Startup);
+        }
+
+        public IList<Counter> Parse(string mergeTraceFile, string processName)
+        {
+            var results = new List<double>();
+            double threadTime = 0;
+            var threadTimes = new List<double>();
+            var ins = new Dictionary<int, double?>();
+            double start = -1;
+            int? pid = null;
+            using (var source = new ETWTraceEventSource(mergeTraceFile))
+            {
+                
+                source.Kernel.ProcessStart += evt =>
+                {
+                    if(evt.ProcessName.Equals(processName, StringComparison.OrdinalIgnoreCase))
+                    {
+                        if(pid.HasValue)
+                        {
+                            throw new Exception("found a start when we already had a start");
+                        }
+                        pid = evt.ProcessID;
+                        start = evt.TimeStampRelativeMSec;
+                    }
+                };
+
+                source.Kernel.ThreadCSwitch += evt =>
```

Comment:
It would be nice to only have one copy of a bunch of this logic since it's the same across parsers and is just complicated enough.  I wouldn't block on this though.

Reply:
Yeah, this refactoring is definitely on my list.

Reply:
Cool.